src/loading-screen/use-loading-state.ts

```diff
@@ -23,15 +24,22 @@ export const useLoadingState = (timeoutMs: number): LoadingState => {
 
   const status = loadSuccessful ? 'success' : isTimeout ? 'timeout' : 'loading';
 
+  useEffect(() => setIsTimeout(false), [userId]);
```

Comment:
Won't timeout be set to false on mount here? Is that intended?

Reply:
Yeah should be fine as is. However I think it does mean that the order of the useEffects matter.

Reply:
Not right now due to the `setIsTimeout(true)` being wrapped with a `setTimeout`. However in general, when updating a state variable on mount, if several useEffects set it, the order will matter.